the-collab-lab/tcl-34-smart-shopping-list
# Save estimated next purchase in Firestore

## Description

Changes:
- `compareDates`: now it returns a day instead of the amount of hours, we did this refactor as we needed this function to compare two dates and store them in firestore. We also modified the consts so they're more human-readable turning them into declarated const values.
- `show-products/index.js`: We search for the item clicked in the State, then we use the `calculateEstimate` function from `@the-collab-lab/shopping-list-utils` and send the neccesary values making sure it receives numbers instead of strings -especially because of the time frames, where they're stored in a string even though they're a number-

## Related Issue

Closes #10 

## Acceptance Criteria

-Cuando se registra una compra, el número estimado de días hasta la próxima fecha de compra debe ser calculado y registrado en la base de datos.

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/37032132/140325824-246f42c1-2a5e-4931-9108-82b21f59953a.png)


### After

![image](https://user-images.githubusercontent.com/37032132/140325911-a68879a2-a510-4537-b131-f00e98e86490.png)

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin mm-yu-estimated-next-purchase` and check that branch out with `git switch mm-yu-estimated-next-purchase`
- `npm start` to launch the app.
- Either add a new item or check one, then go to [Firestore](https://console.firebase.google.com/u/0/project/tcl-34-smart-shopping-list/firestore/) search for your product and see the changes in estimated next purchase
